app/controllers/nodes_controller.rb

```diff
@@ -16,6 +18,18 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def bootstrap
+    salt = Pharos::Salt.new
+    available_roles = %i(master)
+    salt.minions.keys.each do |minion|
```

Comment:
We'd need to accomodate this code after other changes make it inside the repo. I agree about the single source of truth, this is a way of getting the nodes without blocking because of other WIP PR, it will go away.